.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -16,11 +16,17 @@
 
 ## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
 
-## Added to documentation?
+## Pull Request Checklist
 
-- [ ] docs.dev.to
-- [ ] readme
-- [ ] no documentation needed
+I have:
```

Comment:
I wonder if this should be a friendly reminder rather than a checklist 🤔 This is five more checkmarks we have to add to every PR which is a bit of a pain and I also think PR descriptions should really be FOR the PR. This is more of a personal checklist that doesnt add any additional context to anyone else if they were looking at the PR.

Reply:
I agree that this is a lot of checkboxes.

I'm mostly just modeling this after other projects with these patterns.

Alternatively, it'd be cool if we set up some kind of CLA bot for the Code of Conduct. Additionally, I'd really like to get some mention of including tests.

I'm also not sold on using Pull Request descriptions for holding context about a set of changes because they are very impermanent and tied to a vendor. That is visible in this repo where many of our commit messages link to Pull Requests that no longer exist (because the repo once moved), so that context is lost. Which is why I think it's worthwhile to mention commit messages.

Do you think this would be better achieved by pushing contributors to CONTRIBUTING.md where that information can live? Generally, I'm afraid that many contributors miss documentation in those files because they've got to go looking for that info, then read it.

This also isn't something worth spending too much energy on; I'd just like to find a good way to encourage testing for new functionality and catching regressions. That being said, suggestions would be greatly appreciated :smile:

Reply:
What if we add a single line like this and put that checklist as the link?

> DELETE THIS after you have completed the [Pull Request Checklist](https://github.com/thepracticaldev/dev.to/blob/master/CONTRIBUTING.md#create-a-pull-request)

Reply:
What if these were a bulleted reminder in the PR template that's commented out so only the person submitting the PR can see?

Reply:
I like that we want to experiment with this, I also like Jess's suggestion, it will remove a little bit of "fear of public judgment" (don't know how best to explain this). People could still tick all of these without actually having down the things described, so I don't see much value in keeping this list public in the final PR.

We should probably, in a separate PR, add a section about commit messages and commit templates in our contributing guide. For example i use the following template (which I ignore most of the time TBH because I use the PRs to explain the context):

```text
# If applied, this commit will...


# Explain why this change is being made

# Provide links to any relevant tickets, articles or other resources
```